### PR TITLE
[summerospp]add fluentbit opentelemetry plugin

### PR DESCRIPTION
**Pull request by sjliu1**
### Which issue(s) this PR fixes:

Fixes #

### Does this PR introduced a user-facing change?

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by wenchajun — DISMISSED** [review]

**Review by jjsiv — COMMENTED** [review]
Hey, can you check the failing test?

File: apis/fluentbit/v1alpha2/plugins/input/open_telemetry_types.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=65535
+	Port *int32 `json:"port,omitempty"`
+	// Specify the key name to overwrite a tag. If set, the tag will be overwritten by a value of the key.
+	Tagkey string `json:"tagKey,omitempty"`
+	// Route trace data as a log message(default false).
+	RawTraces *bool `json:"rawTraces,omitempty"`
+	// Specify the maximum buffer size in KB to receive a JSON message(default 4M).
+	// +kubebuilder:validation:Pattern:="^\\d+(k|K|KB|kb|m|M|MB|mb|g|G|GB|gb)?$"
+	bufferMaxSize string `json:"bufferMaxSize,omitempty"`
```
Comment (wenchajun) [inline apis/fluentbit/v1alpha2/plugins/input/open_telemetry_types.go:28]:
```suggestion
	BufferMaxSize string `json:"bufferMaxSize,omitempty"`
```

File: apis/fluentbit/v1alpha2/plugins/input/open_telemetry_types.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	if ot.Port != nil {
+		kvs.Insert("port", fmt.Sprint(*ot.Port))
+	}
+	if ot.Tagkey != "" {
+		kvs.Insert("tag_key", ot.Tagkey)
+	}
+	if ot.RawTraces != nil {
+		kvs.Insert("raw_traces", fmt.Sprint(*ot.RawTraces))
+	}
+	if ot.bufferMaxSize != "" {
```
Comment (wenchajun) [inline apis/fluentbit/v1alpha2/plugins/input/open_telemetry_types.go:55]:
```suggestion
	if ot.BufferMaxSize != "" {
```

File: apis/fluentbit/v1alpha2/plugins/input/open_telemetry_types.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		kvs.Insert("buffer_max_size", ot.bufferMaxSize)
```
Comment (wenchajun) [inline apis/fluentbit/v1alpha2/plugins/input/open_telemetry_types.go:56]:
```suggestion
		kvs.Insert("buffer_max_size", ot.BufferMaxSize)
```

**Review by wenchajun — DISMISSED** [review]

File: apis/fluentbit/v1alpha2/clusterinput_types.go [hunk 4]
```diff
@@ -55,6 +55,8 @@ type InputSpec struct {
 	CustomPlugin *custom.CustomPlugin `json:"customPlugin,omitempty"`
 	// Forward defines forward  input plugin configuration
 	Forward *input.Forward `json:"forward,omitempty"`
+	// OpenTelemetry defines forward  input plugin configuration
+	OpenTelemetry *input.OpenTelemetry `json:"opentelemetry,omitempty"`
```
Comment (benjaminhuo) [inline apis/fluentbit/v1alpha2/clusterinput_types.go:59]:
```suggestion
	OpenTelemetry *input.OpenTelemetry `json:"openTelemetry,omitempty"`
```

**Review by wenchajun — APPROVED** [review]

File: config/crd/bases/fluentbit.fluent.io_clusterinputs.yaml [hunk 5]
```diff
@@ -156,6 +156,42 @@ spec:
                       plugin.
                     type: string
                 type: object
+              opentelemetry:
```
Comment (benjaminhuo) [inline config/crd/bases/fluentbit.fluent.io_clusterinputs.yaml:159]: Still opentelemetry , You'll need to regenerate the yaml by `make test`
And the `opentelemetry` in the helm chart needs to be changed to `openTelemetry `

Comment (sjliu1) [reply config/crd/bases/fluentbit.fluent.io_clusterinputs.yaml]: Thanks,  I have done it.

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
Can the docs here https://github.com/fluent/fluent-operator/blob/master/docs/plugins/fluentbit/index.md be generated by `make docs-update` ? @wenchajun

**Comment by wenchajun** [comment]
> Can the docs here https://github.com/fluent/fluent-operator/blob/master/docs/plugins/fluentbit/index.md be generated by `make docs-update` ? @wenchajun

Yes, it can be generated automatically.

**Comment by benjaminhuo** [comment]
> > Can the docs here https://github.com/fluent/fluent-operator/blob/master/docs/plugins/fluentbit/index.md be generated by `make docs-update` ? @wenchajun
> 
> Yes, it can be generated automatically.

@sjliu1 You'll need to update the docs as well

**Comment by sjliu1** [comment]
> > > Can the docs here https://github.com/fluent/fluent-operator/blob/master/docs/plugins/fluentbit/index.md be generated by `make docs-update` ? @wenchajun
> > 
> > 
> > Yes, it can be generated automatically.
> 
> @sjliu1 You'll need to update the docs as well

**Comment by sjliu1** [comment]
> > > Can the docs here https://github.com/fluent/fluent-operator/blob/master/docs/plugins/fluentbit/index.md be generated by `make docs-update` ? @wenchajun
> > 
> > 
> > Yes, it can be generated automatically.
> 
> @sjliu1 You'll need to update the docs as well

> OK, done

**Pull request merged**